src/app/core/global/global.service.ts

```diff
@@ -32,44 +36,65 @@ export class GlobalService {
         });
     }
 
-    checkForPlatformUpdates() {
+    checkForPlatformUpdates(showUpToDateModal: boolean = false) {
 
-        if (!this.checkingForPlatformUpdate) {
+        if (this.checkForPlatformUpdatePromise) {
+            return this.checkForPlatformUpdatePromise;
+        }
 
-            this.checkingForPlatformUpdate = true;
+        this.checkForPlatformUpdatePromise = new Promise((resolve, reject) => {
 
             const process = this.statusBar.startProcess("Checking for platform updates.");
 
-            this.platformRepository.checkForPlatformUpdates().withLatestFrom(this.localRepository.getUpdateAvailable())
+            this.ipc.request("checkForPlatformUpdates").withLatestFrom(this.localRepository.getIgnoredUpdateVersion())
                 .take(1).subscribe((result) => {
 
-                this.checkingForPlatformUpdate = false;
-                this.statusBar.stopProcess(process, "Checking for platform updates is finished.");
+                this.checkForPlatformUpdatePromise = null;
+                this.statusBar.stopProcess(process, "");
 
-                const [hasUpdate, lastAvailableUpdateStored] = result;
+                const [hasUpdate, ignoredUpdateVersion] = result;
 
                 if (!hasUpdate) {
```

Comment:
Flipping the if-else clause to check for positive statement instead of negation would be better style.